Jellyfin.Server/Program.cs

```diff
@@ -647,5 +654,78 @@ private static string NormalizeCommandLineArgument(string arg)
 
             return "\"" + arg + "\"";
         }
+
+        private static void MigrateNetworkSettings(ILogger logger, ServerApplicationPaths appPaths)
```

Comment:
This should really be in a separate file. I also don't really like that it uses _current_ classes. If someone were to apply this migration in 10.8 it might not work as expected. I really want migrations to version-agnostic if possible.

Reply:
You mean a static oldXml.value -> newXml value migration?

Reply:
Pretty much yes. Something that can be run if you're migrating from 10.6 to whatever.

Reply:
Moved migration code into the Jellyfin.Server\Migrations where the rest resides.

Tried static migration but it turned into coding nighmare with pages of string comparisons, so i decided to look at the original code to see what can be done.

For the properties to be absent from `ServerConfiguration`, the file has to be treated as an XML. The callback function pemits us to intercept properties that are no longer in the class, so the migration only works with properties that are not longer valid in `ServerConfiguration`.  If any properties are redefined in `ServerConfiguration` in the future, they will not be migrated.

The second part is not ideal - instead of migrating if network.xml is not present, it only migrates once using the migrate.xml similar to other migration functions. However, since `CommonConfiguration` overwrites system.xml with a 'clean' version at creation, the existing Migration routine cannot be used, so i've had to add the config by hand. (not ideal, but i can't find another solution without altering  lots of other code).

The advantage of this is that all migration results are in one place.

Reply:
If it makes it easier I would just make a `config_migrations.xml`.

And then to make it even simpler I would create two private classes that contains the properties you want to migrate; one for system.xml and one for network.xml. Then utilize the xmlserializer to move it from A to B. The end.

Reply:
One phrase for this one - ball ache!

What i've learnt so far, on what should have been a simple thing.

Serialization classes cannot be private - best they can be is internal.
You cannot typecast on a serialiser.

I had originally thought to declare two classes, the original inheritied from the network - that way properties would only need to declared once. The plan was then to read from source into ServerConfig, and write typecasting. Nope - writing saves the original class' object into the file.

So i've done the next best thing. desearialised into legacy class. Copied properties. Save into network class.